### PR TITLE
NIFI-13811: Fix CalculateRecordStats so that it can properly handle a…

### PR DESCRIPTION
**Pull request by markap14**
…rrays of nested records; clarified documentation / code cleanup

# Summary

[NIFI-00000](https://issues.apache.org/jira/browse/NIFI-00000)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [ ] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [ ] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [ ] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [ ] Pull Request based on current revision of the `main` branch
- [ ] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 21

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/CalculateRecordStats.java [hunk 1]
```diff
@@ -133,104 +141,106 @@ public Set<Relationship> getRelationships() {
...
-            Map<String, String> stats = getStats(input, paths, context, session);
-
-            input = session.putAllAttributes(input, stats);
-
-            session.transfer(input, REL_SUCCESS);
-
-        } catch (Exception ex) {
-            getLogger().error("Error processing stats.", ex);
-            session.transfer(input, REL_FAILURE);
+            final Map<String, RecordPath> recordPath = getRecordPaths(context, flowFile);
```
Comment (EndzeitBegins) [inline nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/CalculateRecordStats.java:150]: Did you use the singular form here on purpose or accidentally? I'd prefer `recordPaths` over `recordPath`.

**Review by EndzeitBegins — APPROVED** [review]
Code changes look good to me. 
Thank you for refactoring some names for improved readability. 

You could use `List.of` for schema declaration in tests instead of `ArrayList`, but that's just a matter of preference.
Other than that I just have one small remark regarding a variable name.

File: nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/CalculateRecordStats.java (hunk 1 above)
Comment (exceptionfactory) [reply nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/CalculateRecordStats.java:150]: That's worth adjusting @EndzeitBegins, thanks.

**Review by exceptionfactory — APPROVED** [review]
Thanks for the fixes and improvements @markap14! The overall set of changes look good, I'll adjust the name of the `recordPath` variable to `recordPaths` in the merge. +1

**Pull request closed without merging**